### PR TITLE
Add scheduler module to library reference.

### LINKED ISSUE #1178: Add Scheduler to Library Reference
The scheduler module is not at all listed in the Library Reference.

### PR DESCRIPTION
**Pull request by cmarqu**
Closes #1178.

### CODE REVIEW

**Comment by eric-wieser** [comment]
I think it's worth clarifying that none of these methods should be used directly in test code - perhaps in a ..note::.

There's an awful lot of fragile methods that probably ought to be private.

File: documentation/source/library_reference.rst [hunk 1]
```diff
@@ -66,6 +66,15 @@ classes used within ``cocotb``.
     :members:
     :member-order: bysource
 
+The Scheduler
+-------------
+
+.. currentmodule:: cocotb.scheduler
+
+.. autoclass:: Scheduler
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst]: Maybe:
```
.. note::
    The scheduler object should generally not be interacted with directly - the only part of it that you'll need is encapsulated in :func:`cocotb.fork`, everything else works behind the scenes.

Comment (eric-wieser) [reply documentation/source/library_reference.rst]: Also perhaps add:
```
.. py:variable:: scheduler
    The global scheduler instance
```
Worth looking up what the correct Sphinx syntax is for that.

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Added this with a docstring on ``scheduler`` and ``.. autodata:: cocotb.scheduler``.

**Comment by eric-wieser** [comment]
close / reopen to rebuild with the fork change merged

File: documentation/source/library_reference.rst [hunk 2]
```diff
@@ -282,3 +281,24 @@ Signal Tracer for WaveDrom
     :members:
     :member-order: bysource
     :synopsis: A signal tracer for WaveDrom.
+
+
+Developer-focused
+=================
```
Comment (eric-wieser) [inline documentation/source/library_reference.rst]: For reviewing: https://external-builds.readthedocs.io/html/cocotb/1179/library_reference.html#developer-focused

Comment (cmarqu) [reply documentation/source/library_reference.rst]: I don't immediately see why ``fork()`` is still not linked... :/

Comment (eric-wieser) [reply documentation/source/library_reference.rst]: It is linked?

Comment (cmarqu) [reply documentation/source/library_reference.rst]: Indeed it is linked, maybe some caching glitch.

**Comment by cmarqu** [comment]
> A little wierd how the docstring for `add` now appears in two places, but this is still clearly an improvement, and I'm not sure there's much that can be done about it anyway.

Maybe adding "``fork()``/``scheduler.add()`` are aliases" makes that clear enough?

**Comment by eric-wieser** [comment]
@cmarqu: It's tempting to either (in another PR):

1. change fork to:
    ```
    def fork(coro):
        """
        long user facing docstring here
        """
        return scheduler.add(coro)
    ```
2. change `Scheduler.add` to `Scheduler._fork`, so that it doesn't appear in the docs for `Scheduler` at all, and is instead just accessible only through `cocotb.fork`
3. Do the above but without the underscore, which at least aligns the names
4. Eliminate `Scheduler.add` entirely, and move its contents to `cocotb.fork`. It's not clear to me why the scheduler itself uses `self.add` rather than `self.schedule`. It's something I want to look into, but i'm waiting on #1173 which is waiting on a review from @imphil before I investigate further.

**Comment by eric-wieser** [comment]
I'd merge this, but I'm waiting for the merge of #1174 before merging anything non-trivial

**Comment by eric-wieser** [comment]
Nevermind, this is definitely under the realm of "trivial"

**Comment by eric-wieser** [comment]
Thanks @cmarqu!

**Pull request merged**